marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -256,23 +253,40 @@ constructMarloweType constructorName (MarloweHole { row, column }) m = case Map.
       let
         newArgs = getMarloweConstructors marloweType
 
-        hole = MarloweHole { name, marloweType, row: 0, column: 0 }
+        hole = MarloweHole { name, marloweType, range: emptyRange }
       in
         constructMarloweType name hole newArgs
 
   parens 1 1 text = text
 
   parens _ _ text = "(" <> text <> ")"
 
-mkHole :: forall a. String -> { row :: Pos, column :: Pos } -> Term a
-mkHole name position = Hole name Proxy position
+type Range
+  = { startLineNumber :: Int
+    , startColumn :: Int
+    , endLineNumber :: Int
+    , endColumn :: Int
+    }
 
-mkDefaultTerm :: forall a. a -> Term a
-mkDefaultTerm a = Term a { row: 0, column: 0 }
+emptyRange :: Range
+emptyRange =
+  { startLineNumber: 0
+  , startColumn: 0
+  , endLineNumber: 0
+  , endColumn: 0
+  }
```

Comment:
I don't know if you want to, but I'll bet you could use `emptyRange = zero` instead.

Reply:
oh nice